receiver/dockerstatsreceiver/receiver.go

```diff
@@ -293,9 +295,35 @@ func calculateCPUPercent(previous *dtypes.CPUStats, v *dtypes.CPUStats) float64
 	return cpuPercent
 }
 
-func calculateMemoryPercent(memoryStats *dtypes.MemoryStats) float64 {
-	if float64(memoryStats.Limit) == 0 {
-		return 0
+// calculateMemUsageNoCache calculate memory usage of the container.
```

Comment:
Since you've taken this function from the Docker CLI code, which is under the Apache license, it will need a copyright notice similar to `calculateCPUPercent()`.

Reply:
I have updated the notice that is on top of `calculateCPUPercent` which is copied from the same place to avoid having this notice multiple times, perhaps I can move all these three functions `calculateCPUPercent`, `calculateMemUsageNoCache`, and `calculateMemoryPercent` to a single file with one notice in the top WDYT?

Reply:
I like that idea

Reply:
done in [6f3b680](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21166/commits/6f3b680abdecfa295ede3e81d0c61909a13c61c7)